### PR TITLE
feat(onClickOutside): default to just pointerDown 

### LINKED ISSUE #451: potential bug: onClickOutside
It looks like `onClickOutside` is being called multiple times when the element target is not removed after the `onClickOutside` event is fired. This doesn't show up on the VueUse demo page because the targets are removed, but if they aren't it will be fired twice.

Here is a reproduction on [Vue SFC Playground](https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IG9uQ2xpY2tPdXRzaWRlIH0gZnJvbSAnQHZ1ZXVzZS9jb3JlJ1xuXG5jb25zdCB0YXJnZXQgPSByZWYoKVxuY29uc3QgY2xpY2tlZCA9IHJlZigwKVxuXG5vbkNsaWNrT3V0c2lkZSh0YXJnZXQsICgpID0+IGNsaWNrZWQudmFsdWUgKz0gMSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXYgc3R5bGU9XCJkaXNwbGF5OiBncmlkOyBwbGFjZS1jb250ZW50OiBjZW50ZXI7IHBsYWNlLWl0ZW1zOiBjZW50ZXI7IGhlaWdodDogMTAwJTsgZ2FwOiAxcmVtOyB1c2VyLXNlbGVjdDogbm9uZTtcIj5cbiAgICBDbGlja2VkIE91dHNpZGUge3sgY2xpY2tlZCB9fSB0aW1lKHMpXG4gICAgPGRpdiByZWY9XCJ0YXJnZXRcIiBzdHlsZT1cImJhY2tncm91bmQ6IHdoZWF0OyBwYWRkaW5nOiAycmVtOyBib3JkZXItcmFkaXVzOiAwLjI1cmVtO1wiPlxuICAgICAgQ2xpY2sgT3V0c2lkZSBvZiBtZVxuICAgIDwvZGl2PlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG5cdFx0XCJAdnVldXNlL2NvcmVcIjogXCJodHRwczovL3VucGtnLmNvbS9AdnVldXNlL2NvcmUvZGlzdC9pbmRleC5lc20uanNcIixcbiAgICBcIkB2dWV1c2Uvc2hhcmVkXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQHZ1ZXVzZS9zaGFyZWQvZGlzdC9pbmRleC5lc20uanNcIixcbiAgICBcInZ1ZS1kZW1pXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vdnVlLWRlbWkvbGliL2luZGV4LmVzbS5qc1wiXG4gIH1cbn0ifQ==)
 
Is this behavior intentional? Maybe this behavior was introduced in this issue https://github.com/vueuse/vueuse/issues/323? If this is intentional maybe we should add an option to only fire the event once.

### PR DESCRIPTION
**Pull request by wheatjs**
default to just pointerDown and add option to include more events. Closes #451

### CODE REVIEW

File: packages/core/onClickOutside/index.ts [hunk 1]
```diff
@@ -6,6 +6,10 @@ import { ConfigurableWindow, defaultWindow } from '../_configurable'
 const events = ['mousedown', 'touchstart', 'pointerdown'] as const
 type EventType = WindowEventMap[(typeof events)[number]]
 
+export interface OnClickOutsideOptions extends ConfigurableWindow {
+  events: ('mousedown' | 'touchstart' | 'pointerdown')[]
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:10]:
```suggestion
  events: string[]
```

File: packages/core/onClickOutside/index.ts [hunk 2]
```diff
@@ -17,7 +21,9 @@ type EventType = WindowEventMap[(typeof events)[number]]
 export function onClickOutside(
   target: MaybeElementRef,
   handler: (evt: EventType) => void,
-  options: ConfigurableWindow = {},
+  options: OnClickOutsideOptions = {
+    events: ['pointerdown'],
+  },
 ) {
   const { window = defaultWindow } = options
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:28]: Let's assign the default value here.

**Review by antfu — CHANGES_REQUESTED** [review]

File: packages/core/onClickOutside/index.ts (hunk 2 above)
Comment (wheatjs) [reply packages/core/onClickOutside/index.ts:28]: Which part do you mean?

**Comment by sibbng** [comment]
I my opinion we should only accept single string of whitelisted event names. Contrary to my earlier suggestion I don't think uses cases which needs to listen multiple pointer events are that common. 

Also to return proper type on event handler, end users have to mark their events array with `as const` which is a extra overhead in my opinion. As long as current status of this PR only accept array, it is also an overhead for anyone who needs to listen only one event.

**Comment by wheatjs** [comment]
> I my opinion we should only accept single string of whitelisted event names. Contrary to my earlier suggestion I don't think uses cases which needs to listen multiple pointer events are that common.
> 
> Also to return proper type on event handler, end users have to mark their events array with `as const` which is a extra overhead in my opinion. As long as current status of this PR only accept array, it is also an overhead for anyone who needs to listen only one event.

So we should only listen to `pointerdown` then?

File: packages/core/onClickOutside/index.ts (hunk 2 above)
Comment (antfu) [reply packages/core/onClickOutside/index.ts:28]:
```ts
 const { window = defaultWindow, events = ['pointerdown'] } = options
```

**Comment by sibbng** [comment]
I pushed the changes I suggested above. Also it gives proper event type to handler based on `event` option. I'm happy with this changes.

**Comment by wheatjs** [comment]
This looks good to me

File: packages/core/onClickOutside/index.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
-import { Fn, tryOnUnmounted } from '@vueuse/shared'
+import { tryOnUnmounted } from '@vueuse/shared'
 import { MaybeElementRef, unrefElement } from '../unrefElement'
 import { useEventListener } from '../useEventListener'
 import { ConfigurableWindow, defaultWindow } from '../_configurable'
 
-const events = ['mousedown', 'touchstart', 'pointerdown'] as const
-type EventType = WindowEventMap[(typeof events)[number]]
+export type AllowedEvents = Pick<WindowEventMap, 'mousedown' | 'mouseup' | 'touchstart' | 'touchend' | 'pointerdown' | 'pointerup'>
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:6]:
```suggestion
export type OnClickOutsideAllowedEvents = Pick<WindowEventMap, 'mousedown' | 'mouseup' | 'touchstart' | 'touchend' | 'pointerdown' | 'pointerup'>
```

Comment (sibbng) [reply packages/core/onClickOutside/index.ts:6]: I knew it 😂 What about `OnClickOutsideEvents`? We are using it a lot on code and this is little bit longer.

Comment (antfu) [reply packages/core/onClickOutside/index.ts:6]: Sure.

**Review by sibbng — APPROVED** [review]

**Review by antfu — APPROVED** [review]

**Pull request merged**